packages/hash/api/src/model/knowledge/user.model.ts

```diff
@@ -158,13 +164,28 @@ export default class extends EntityModel {
       );
     }
 
+    if (shortname) {
+      if (AccountFields.shortnameIsInvalid(shortname)) {
+        throw new Error(`The shortname "${shortname}" is invalid`);
+      }
+
+      if (
+        AccountFields.shortnameIsRestricted(shortname) ||
+        (await AccountFields.shortnameIsTaken(graphApi, { shortname }))
+      ) {
+        throw new Error(
+          `A user entity with shortname "${shortname}" already exists.`,
+        );
+      }
```

Comment:
Presumably this will need to check for the existence of an org with the same shortname also, since they share URLs (`https://hash.ai/@[namespace]`

Reply:
Correct 👍

And that's what [the implementation of `AccountFields.shortnameIsTaken`](https://github.com/hashintel/hash/blob/55d1c963425fc76a87544792ba270468ad74b649/packages/hash/api/src/model/knowledge/account.fields.ts#L26-L40) is doing

Reply:
Yes, thanks for flagging - have addressed in https://github.com/hashintel/hash/pull/1360/commits/00f2d5cbc0024e7c285882df1e9261c20d7000fb

Reply:
Have found the same mistake in some other places, which I've addressed in https://github.com/hashintel/hash/pull/1360/commits/6c04d09a5df87c5928aed74f6619e7ad5a4818d9